lib/multi.c

```diff
@@ -367,6 +368,21 @@ struct Curl_multi *Curl_multi_handle(int hashsize, /* socket hash */
 
   /* -1 means it not set by user, use the default value */
   multi->maxconnects = -1;
+
+#ifdef ENABLE_UNBLOCK
+  if(Curl_socketpair(AF_UNIX, SOCK_STREAM, 0, multi->unblock_pair) < 0) {
```

Comment:
- wouldn't that be confusing to have AF_UNIX even on windows? (I know that it's ignored inside Curl_socketpair)
 - why sock_stream, udp should to be lighter than tcp.

Reply:
> wouldn't that be confusing to have AF_UNIX even on windows? (I know that it's ignored inside Curl_socketpair)

Won't it be equally confusing to setup an #ifdef for the value when it isn't even used on that platform?

> why sock_stream? udp should to be lighter than tcp

Well, we want the pipe to be a reliable stream. I'm not sure what to expect from this if we use datagrams instead... besides, does it really matter (can the "overhead" even be measured?) for this simple use case?

Reply:
I chose these parameters because this is how it is already used in the threaded resolver. These can be later improved in both places in a separate PR. Based on https://stackoverflow.com/questions/13953912/difference-between-unix-domain-stream-and-datagram-sockets I think SOCK_STREAM is better here because if we always send 1-byte messages then we would only be able to read a single byte at a time with SOCK_DGRAM (Note: I didn't test if this is really the case)